MetaMask/metamask-extension
# fix: hackily wait longer for linea swap approval tx to increase chance of success

## **Description**

Linea seems to be taking longer than other chains to process the approve transaction after it is submitted, so the trade transaction is erroring. This PR adds a hacky workaround where we artificially delay if we're on Linea to give the trade transaction more time. In the future, we'd want to avoid this hack, but for now it should increase the swap success rate on Linea.

With the delay, the token symbol also wasn't immediately populating on the awaiting swap page, so this PR also updates how it's retrieved.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27810?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27804

## **Manual testing steps**

1. Start a swap on Linea with a token that you have not granted approval for
2. Observe that the swap does not fail

## **Screenshots/Recordings**

### **Before**

https://github.com/user-attachments/assets/57cdc5e5-cea7-48ad-ba13-38820ecc9155

### **After**

https://github.com/user-attachments/assets/91bbfbf4-8392-41ea-bfe8-d54813758f5c



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
